File: perl/knight.pl
#!/usr/bin/perl

use strict;
use warnings;

use lib 'lib';
use Kn;

# Prints a usage message and exists
sub usage {
	print STDERR "usage: $0 (-e program | -f file)\n";
	exit 1
}

my $sw = shift @ARGV;

usage unless $#ARGV == 0 && ($sw eq '-f' || $sw eq '-e');

if ($sw eq '-f') {
	Kn->run(join '', <>);
} elsif ($sw eq '-e') {
	Kn->run(shift @ARGV);
} else {
	usage;
}

File: php/bin/knight.php
#!/usr/bin/env php
<?php

include_once __DIR__ . '/../src/Knight.php';

switch(count($argv) === 3 ? $argv[1] : null) {
	case '-e':
		\Knight\run($argv[2]);
		break;
	case '-f':
		\Knight\run(file_get_contents($argv[2]));
		break;
	default:
		die("USAGE: $argv[0] (-e 'code' | -f file)\n");
}
